sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -422,6 +432,7 @@ class StreamExecution(
     val replacementMap = AttributeMap(replacements)
     val triggerLogicalPlan = withNewSources transformAllExpressions {
       case a: Attribute if replacementMap.contains(a) => replacementMap(a)
+      case t: CurrentTimestamp => CurrentBatchTimestamp(currentBatchTimestamp)
```

Comment:
Good question. The purpose of the dedicated expression is to avoid having it pushed down below a stateful operator (e.g., aggregation). during optimization.